src/vos/vos_iterator.c

```diff
@@ -532,13 +532,31 @@ need_reprobe(vos_iter_type_t type, struct vos_iter_anchors *anchors)
 	return reprobe;
 }
 
+int
+vos_iter_aggregate(daos_handle_t ih, bool discard)
+{
+	struct vos_iterator *iter = vos_hdl2iter(ih);
+	int rc;
+
+	rc = iter_verify_state(iter);
+	if (rc)
+		return rc;
```

Comment:
iter_verify_state() can return -DER_NONEXIST to indicate the end of the iteration, seems different from the comment "-DER_NONEXIST if the entry has been removed entirely". Is it expected?